hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/ScheduleCompactionActionExecutor.java

```diff
@@ -140,6 +140,17 @@ private Option<Pair<Integer, String>> getLatestDeltaCommitInfo() {
     return Option.empty();
   }
 
+  private Option<Pair<Integer, String>> getLatestDeltaCommitInfoSinceLastCompactionRequest() {
+    Option<Pair<HoodieTimeline, HoodieInstant>> deltaCommitsInfo =
+          CompactionUtils.getDeltaCommitsSinceLatestCompactionRequest(table.getActiveTimeline());
+    if (deltaCommitsInfo.isPresent()) {
+      return Option.of(Pair.of(
+            deltaCommitsInfo.get().getLeft().countInstants(),
```

Comment:
So the compaction scheduling would be affected by the progress of the compaction executions ? Is that reasonable ? Don't think so.

Reply:
For now, NUM_COMMITS or TIME_ELAPSED compaction trigger strategy will check the number or time of delta-commits after the last successful compaction. so if the offline compaction application crash for a while(or async compaction is very slow), there will be a lot of compaction request（one request per delta commit） in the timeline, and that will have a side effect on performance.

so this PR provides a new strategy not to check the last successful compaction but check the last compaction request if possible.

Reply:
@danny0405 Any input or question?